x-pack/test/api_integration/apis/entity_manager/search.ts

```diff
@@ -0,0 +1,456 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import moment from 'moment';
+import expect from 'expect';
+import { FtrProviderContext } from '../../ftr_provider_context';
+import { createEntitySourceDefinition, searchEntities } from './helpers/request';
+import { createIndexWithDocuments } from './helpers/data_generation';
+
+export default function ({ getService }: FtrProviderContext) {
+  const esClient = getService('es');
+  const supertest = getService('supertest');
+
+  describe('_search API', () => {
+    it('returns 404 when no matching sources', async () => {
+      await searchEntities(supertest, { type: 'undefined-type' }, 404);
```

Comment:
I'm on the fence for this one because I think it would be good to verify integrity of type + sources but we're in a path that needs to be performant and that extra check does not really add value, so I'd say only sources check is fine